src/Storages/addColumnsStructureToQueryWithClusterEngine.cpp

```diff
@@ -0,0 +1,47 @@
+#include <Storages/addColumnsStructureToQueryWithClusterEngine.h>
+#include <Parsers/ASTExpressionList.h>
+#include <Parsers/ASTSelectQuery.h>
+#include <Parsers/ASTTablesInSelectQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/queryToString.h>
+
+namespace DB
+{
+
+static ASTExpressionList * extractTableFunctionArgumentsFromSelectQuery(ASTPtr & query)
+{
+    auto * select_query = query->as<ASTSelectQuery>();
+    if (!select_query || !select_query->tables())
+        return nullptr;
+
+    auto * tables = select_query->tables()->as<ASTTablesInSelectQuery>();
+    auto * table_expression = tables->children[0]->as<ASTTablesInSelectQueryElement>()->table_expression->as<ASTTableExpression>();
+    if (!table_expression->table_function)
+        return nullptr;
+
+    auto * table_function = table_expression->table_function->as<ASTFunction>();
+    return table_function->arguments->as<ASTExpressionList>();
+}
+
+void addColumnsStructureToQueryWithClusterEngine(ASTPtr & query, const String & structure, size_t max_arguments, const String & function_name)
+{
+    /// Add argument with table structure to s3Cluster table function in select query.
```

Comment:
```suggestion
    /// Add argument with table structure to s3Cluster/hdfsCluster table function in select query.
```
Or just `cluster` in both comments

Reply:
I just forgot to delete this comment since I added comment in `.h` file